# --- SDE-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
#
# Filename: package/.../clucene/clucene-gcc-atomics.patch
# Copyright (C) 2013 The OpenSDE Project
#
# More information can be found in the files COPYING and README.
#
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- SDE-COPYRIGHT-NOTE-END ---

--- clucene-core-2.3.3.4/src/shared/CLucene/config/threads.cpp	2012-02-23 12:06:55.355506304 +0000
+++ clucene-core-2.3.3.4/src/shared/CLucene/config/threads.cpp	2012-02-23 12:07:17.131766381 +0000
@@ -185,7 +185,7 @@
     }

     int32_t atomic_threads::atomic_increment(_LUCENE_ATOMIC_INT *theInteger){
-      #ifdef _CL_HAVE_GCC_ATOMIC_FUNCTIONS
+      #if ( __GNUC__ > 4 ) || (( __GNUC__ == 4)  && ( __GNUC_MINOR__ >= 4 ))
         return __sync_add_and_fetch(theInteger, 1);
       #else
         SCOPED_LOCK_MUTEX(theInteger->THIS_LOCK)
@@ -193,7 +193,7 @@
       #endif
     }
     int32_t atomic_threads::atomic_decrement(_LUCENE_ATOMIC_INT *theInteger){
-      #ifdef _CL_HAVE_GCC_ATOMIC_FUNCTIONS
+      #if ( __GNUC__ > 4 ) || (( __GNUC__ == 4)  && ( __GNUC_MINOR__ >= 4 ))
         return __sync_sub_and_fetch(theInteger, 1);
       #else
         SCOPED_LOCK_MUTEX(theInteger->THIS_LOCK)
--- clucene-core-2.3.3.4/src/shared/CLucene/LuceneThreads.h	2012-02-23 12:36:28.388299322 +0000
+++ clucene-core-2.3.3.4/src/shared/CLucene/LuceneThreads.h	2012-02-23 12:37:23.131885433 +0000
@@ -70,7 +70,7 @@
			void NotifyAll();
					};

-          #ifdef _CL_HAVE_GCC_ATOMIC_FUNCTIONS
+          #if ( __GNUC__ > 4 ) || (( __GNUC__ == 4)  && ( __GNUC_MINOR__ >= 4 ))
             #define _LUCENE_ATOMIC_INT uint32_t
             #define _LUCENE_ATOMIC_INT_SET(x,v) x=v
             #define _LUCENE_ATOMIC_INT_GET(x) x
